### PR TITLE
Refactors

### PR DESCRIPTION
**Pull request by bubchi89**

### CODE REVIEW

File: src/auto_import.js [hunk 1]
```diff
@@ -109,34 +106,32 @@ function importFromAPI(urlForAPI) {
...
-        break;
-      }
-
       bannerName = WISH_TALLY_NAME_OF_WISH_HISTORY[i];
       bannerSettings = AUTO_IMPORT_BANNER_SETTINGS_FOR_IMPORT[bannerName];
       if (bannerSettings.is_toggled(settingsSheet)) {
         bannerSheet = SpreadsheetApp.getActive().getSheetByName(bannerName);
         if (bannerSheet) {
-          checkPages(
+          var encounteredErrorCode = checkPages(
```
Comment (FichteFoll) [inline src/auto_import.js]: Ususally, functions report `true` in the success case and not vice versa, but with the variable name it's clean enough.

This is definitely a very valuable refactoring though. Unnecessarily using globals for error tracking is very spaghetti and prone to errors.

File: src/auto_import.js [hunk 2]
```diff
@@ -118,15 +118,15 @@ function importFromAPI(urlForAPI) {
...
+          var success = checkPages(
             bannerSheet,
             bannerName,
             bannerSettings,
             languageSettings,
             selectedServer,
             settingsSheet,
             authKey);
-          if (encounteredErrorCode) {
+          if (success) {
```
Comment (FichteFoll) [inline src/auto_import.js]: This check also needs to be inverted.

**Review by FichteFoll — CHANGES_REQUESTED** [review]

**Review by FichteFoll — APPROVED** [review]
I only did a shallow review since I don't know the code base but was somewhat interested in what the "refactoring" was about, but this generally looks good to me. Make of that as you will.

By the way, I can't resolve the discussions I created because I don't have write access to the branch (for obvious reasons).

**Comment by robqliu** [comment]
Cool, thanks for the review. Yippy said they'd take a look but their hands are full with their work on warp tally. Not super important either way, but figured it might be nice if there are subsequent changes 🤷‍♂️

**Comment by bubchi89** [comment]
Cool, thanks for the review. Yippy said they'd take a look but their hands are full with their work on warp tally. Not super important either way, but figured it might be nice if there are subsequent changes 🤷‍♂️

File: .clasp.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
-    "scriptId": "insert_google_script_id_here",
+    "scriptId": "1HYV03d9Vhm0nrOKz3WgDWOmbCKV0w0cbq3yiXAuuMZerjriqt6rGB108",
```
Comment (Yippy) [inline .clasp.json]: Could you remove your google script id from here, with "insert_google_script_id_here"

File: src/config.js [hunk 4]
```diff
@@ -12,29 +12,200 @@ var WISH_TALLY_SHEET_SCRIPT_IS_ADD_ON = false;
...
 var AUTO_IMPORT_URL_FOR_API_BYPASS = ""; // Optional
+
+class BannerSettings {
+  constructor(range_status_cell, is_toggled_cell, gacha_type) {
+    this.range_status_cell = range_status_cell;
+    this.is_toggled_cell = is_toggled_cell;
+    this.gacha_type = gacha_type;
+  }
+
+  range_status(settingsSheet) {
```
Comment (Yippy) [inline src/config.js:23]: Nice to use classes for setting Import, can the method be renamed to getRangeStatus(settingsSheet), obviously change throughout

File: src/config.js [hunk 5]
```diff
@@ -12,29 +12,200 @@ var WISH_TALLY_SHEET_SCRIPT_IS_ADD_ON = false;
...
+    return settingsSheet.getRange(this.range_status_cell).getValue();
+  }
+
+  set_range_status(value, settingsSheet) {
```
Comment (Yippy) [inline src/config.js:27]: Amend setRangeStatus(value, settingsSheet)

File: src/config.js [hunk 6]
```diff
@@ -12,29 +12,200 @@ var WISH_TALLY_SHEET_SCRIPT_IS_ADD_ON = false;
...
+    return settingsSheet.getRange(this.range_status_cell).setValue(value);
+  }
+
+  is_toggled(settingsSheet) {
```
Comment (Yippy) [inline src/config.js:31]: Amend method to isToggled(settingsSheet) or isEnabled()

File: src/config.js (hunk 4 above)
Comment (Yippy) [reply src/config.js:23]: Maybe a better method name, getStatusText()

File: src/config.js (hunk 5 above)
Comment (Yippy) [reply src/config.js:27]: Maybe a better method name setStatusText()

**Comment by Yippy** [comment]
Overall much better, stripping out the secondary checks that was unnecessary. I will have to move this to Warp Tally later. Also need to add 'AUTO_IMPORT_URL_ERROR_CODE_VISIT_TOO_FREQUENTLY ' that is a new fix in Warp Tally... I will sort that out afterwards.

**Review by Yippy — APPROVED** [review]
Thank you for your PR, there be changes for Genshin Impact v4.5 Chronicled Wish Banners soon.

**Pull request merged**